tests/test_objects.py

```diff
@@ -1021,6 +1022,217 @@ def test_get_object_latest_when_indexd_returns_404_and_guid_not_in_mds(
     assert resp.status_code == 404, resp.text
 
 
+def set_up_metadata_objects(client_fixture):
+    """
+    Create metadata objects from tests/sample_objects.json file.
+    """
+    with open("tests/sample_objects.json") as json_file:
+        data = json.load(json_file)
+
+    for guid, mds_object in data.items():
+        client_fixture.post("/metadata/" + guid, json=mds_object).raise_for_status()
+
+
+def tear_down_metadata_objects(client_fixture):
+    """
+    Delete metadata objects that were created from tests/sample_objects.json
+    file.
+    """
+    with open("tests/sample_objects.json") as json_file:
+        data = json.load(json_file)
+
+    for guid in data.keys():
+        client_fixture.delete("/metadata/" + guid).raise_for_status()
+
+
+def test_get_objects_filter_by_message_equals(client):
+    """
+    Test that the GET /objects endpoint filters correctly when a message-equal
+    filter is applied.
+    """
+    try:
+        set_up_metadata_objects(client)
+
+        resp = client.get('/objects?data=true&filter=(message,:eq,"morning")')
+        resp_json = resp.json()
+
+        assert resp.status_code == 200
+        assert len(resp_json["items"]) == 1
+        assert resp_json["items"][0]["metadata"]["message"] == "morning"
+    finally:
+        tear_down_metadata_objects(client)
```

Comment:
maybe you could use a fixture to avoid adding a try/finally to each test?
maybe something like [this](https://stackoverflow.com/a/22638709/12135968), without  "autouse"?